src/gui/widgets/chatbox.cpp

```diff
@@ -754,6 +754,15 @@ void chatbox::process_message(const ::config& data, bool whisper /*= false*/)
 			room = "lobby";
 		}
 
+		if(log_ != nullptr && data["type"].str() == "motd") {
+			if(log_->at("lobby").received_motd) {
```

Comment:
This should check that it's actually the same message. You could make `received_motd` into a string.

Reply:
Updated.